transaction/std/src/transaction_builder.rs

```diff
@@ -81,49 +80,56 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
     /// Initializes a new TransactionBuilder.
     ///
     /// # Arguments
+    /// * `block_version` - The block version rules to use when building this
+    ///   transaction
+    /// * `fee` - The fee (and token id) to use for this transaction. Note: The
+    ///   fee token id cannot be changed later, and before mixed transactions
+    ///   feature, every input and output must have this token id.
     /// * `fog_resolver` - Source of validated fog keys to use with this
     ///   transaction
     /// * `memo_builder` - An object which creates memos for the TxOuts in this
     ///   transaction
     pub fn new<MB: MemoBuilder + 'static + Send + Sync>(
         block_version: BlockVersion,
-        token_id: TokenId,
+        fee: Amount,
         fog_resolver: FPR,
         memo_builder: MB,
-    ) -> Self {
-        TransactionBuilder::new_with_box(
-            block_version,
-            token_id,
-            fog_resolver,
-            Box::new(memo_builder),
-        )
+    ) -> Result<Self, TxBuilderError> {
+        TransactionBuilder::new_with_box(block_version, fee, fog_resolver, Box::new(memo_builder))
     }
 
     /// Initializes a new TransactionBuilder, using a Box<dyn MemoBuilder>
     /// instead of statically typed
     ///
     /// # Arguments
+    /// * `block_version` - The block version rules to use when building this
+    ///   transaction
+    /// * `fee` - The fee (and token id) to use for this transaction. Note: The
+    ///   fee token id cannot be changed later, and before mixed transactions
+    ///   feature, every input and output must have this token id.
     /// * `fog_resolver` - Source of validated fog keys to use with this
     ///   transaction
     /// * `memo_builder` - An object which creates memos for the TxOuts in this
     ///   transaction
     pub fn new_with_box(
         block_version: BlockVersion,
-        token_id: TokenId,
+        fee: Amount,
         fog_resolver: FPR,
-        memo_builder: Box<dyn MemoBuilder + Send + Sync>,
-    ) -> Self {
-        TransactionBuilder {
+        mut memo_builder: Box<dyn MemoBuilder + Send + Sync>,
+    ) -> Result<Self, TxBuilderError> {
+        // make sure that the memo builder
+        // is initialized to the same fee as the transaction builder
+        memo_builder.set_fee(fee)?;
```

Comment:
Should this be `self.set_fee(fee)?` - or rather, why does `self.set_fee` exist if the fee cannot be changed later, and we are calling `set_fee` directly from the `memo_builder` here?

(Still reading - so I may answer this question for myself)

Reply:
the fee value can be changed, because that was how Matt originally set it up, and that API is exposed and used, but the fee token id cannot be changed